### ISSUE TITLE
Data Templates show as data: {}  for climate.set_temperature and input_number.set_value services when Edit as YAML.

### ISSUE THREAD

**Issue by gb53smith** [home-assistant/frontend#8605]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Accidentally triggered conversion of automations.yaml when I created a new automation in the UI for the first time.  Automations still work except that I can not edit the data templates for climate.set_temperature and input_number.set_value services.  When I try "edit as YAML" I see ```data: {}```.  (The data template worked for the light.turn_on service.)

### Describe the behavior you expected

I expect to see the template expression when I "Edit as YAML".  My workaround is to use NotePad to edit the automations.yaml file.  This defeats the purpose of having an Automations UI.

### Steps to reproduce the issue

Here is a simple automation example:
```
- alias: Last House Humidity
  initial_state: true
  trigger:
    platform: state
    entity_id: sensor.multisensor2_humidity
  action:
  - service: input_number.set_value
    data:
      entity_id: input_number.last_house_humidity
      value: '{{ states(''sensor.multisensor2_humidity'')}}'
```

for climate.set_temperature here are two actions that have this issue:
```
- id: d5c8e48aa51549caa1d295a0a4c0b370
  alias: Home
  description: '# Return temperature set point when returning

    # between heat ON and OFF times if not on vacation'
  trigger:
  - platform: state
    entity_id: binary_sensor.away
    to: 'off'
  condition:
  - condition: template
    value_template: "{{ (now().hour * 60 + now().minute) * 60  \n    > state_attr('input_datetime.heat_on_time',\
      \ 'timestamp') and \n    (now().hour * 60 + now().minute) * 60 \n    < state_attr('input_datetime.heat_off_time',\
      \ 'timestamp') }}\n"
  action:
  - service: climate.set_temperature
    data:
      entity_id: climate.house
      temperature: '{{ states(''input_number.slider_home'')}}'
  - service: climate.set_temperature
    data:
      entity_id: climate.summer_fan
      temperature: !secret temperature_fan
  initial_state: true
  mode: single
```
If I resave the automation my data gets replaced with {}, losing the template expression.  Also !secret is removed and replaced with its value.

### What version of Home Assistant Core has the issue?

2021.3.2

### In which browser are you experiencing the issue with?

Goolge Chrome Version 88.0.4324.190

### Which operating system are you using to run this browser?

Windows 10.0.19041

### State of relevant entities

```yaml
# Paste your state here.

```

### Problem-relevant frontend configuration

```yaml
# Paste your YAML here.

```

### Javascript errors shown in your browser console/inspector

```txt
# Paste your logs here.

```

I could separate out the automations that don't edit into separate directory but prefer to keep everything is one place.

**Comment by bramkragten** [comment]
This is fixed already

**Comment by gb53smith** [comment]
I can confirm that edit of yaml in the UI has been fixed in 2021.3.3 that was released while I was reporting the bug.

The support of !secret is still an issue for me.  I use it as a global variable.  I think I will have to use input number entities.  There values are not shown in the yaml.
